### PR TITLE
[receiver/sqlquery] support attributes for logs

### PR DESCRIPTION
**Pull request by Grandys**
**Description:** Introduced a new configuration option, attribute_columns, for log queries. This option was already available for metric queries.

Note: If an attribute column is missing in the result set, the collection process fails with an error to maintain consistency with metric query behaviour. 

Conversely, in the existing implementation, if the body column is absent, log body will be populated with an empty string. For consistency, it might be preferable for both behaviours to align. However, since this could introduce a breaking change, it may be more appropriate to implement this adjustment in a separate PR.

**Link to tracking Issue:** [24459](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/24459)

**Testing:** Added column `attribute` in test set, and verification in the integration test

**Documentation:** Migrated description for `attribute_columns` from metrics queries section to general "Queries" section, updated example configuration

### CODE REVIEW

**Review by crobert-1 — COMMENTED** [review]
Could we add a test to confirm an error is returned when there is no column for the configured `attribute_columns`?

File: receiver/sqlqueryreceiver/testdata/config-logs.yaml
Comment (crobert-1) [inline receiver/sqlqueryreceiver/testdata/config-logs.yaml:12]:
```suggestion
```
Nit: Do we need the empty line here?

**Review by crobert-1 — APPROVED** [review]

File: receiver/sqlqueryreceiver/testdata/config-logs.yaml
Comment (Grandys) [reply receiver/sqlqueryreceiver/testdata/config-logs.yaml]: We do not, change applied

**Comment by Grandys** [comment]
@crobert-1 is there anything else needed to merge this PR?

**Comment by crobert-1** [comment]
> @crobert-1 is there anything else needed to merge this PR?

It would be nice to get another review here, but nothing else that's blocking. I'll add the `ready to merge` label to let project maintainers know it's ready.

**Review by MovieStoreGuy — APPROVED** [review]

**Pull request merged**
